cleanup tmp name generation

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25065 cleanup tmp name generation**

Using global atomic variables is bad because sending the same AST through
the compiler twice will produce different graphs. This makes it a
member of the translation struct.

Differential Revision: [D16975355](https://our.internmc.facebook.com/intern/diff/D16975355)

zdevito committed:
cleanup tmp name generation

Using global atomic variables is bad because sending the same AST through
the compiler twice will produce different graphs. This makes it a
member of the translation struct.

eellison commented on torch/csrc/jit/script/compiler.cpp:
```diff
@@ -522,6 +522,10 @@ struct to_ir {
   std::vector<DefContext> def_stack_;
+  size_t temp_name_count_ = 0;
+  std::string createTempName(const std::string& prefix) {
+    return prefix + std::to_string(temp_name_count_);
```
i think you dropped a `++`

eellison commented on torch/csrc/jit/script/compiler.cpp:
```diff
@@ -1037,10 +1041,7 @@ struct to_ir {
-    static std::atomic<size_t> tmp_count{0};
-    const auto tmp_name =
-        std::string("___list_acc") + std::to_string(tmp_count++);
+    const auto tmp_name = createTempName("___list_acc");
```
unrelated to your diff but it's better practice to prefix the names with `$` so that there is no possibility of user collision

eellison reviewed (approved):
looks good

zdevito committed:
Update on "cleanup tmp name generation"

cleanup tmp name generation

Using global atomic variables is bad because sending the same AST through
the compiler twice will produce different graphs. This makes it a
member of the translation struct.

gh-metadata: pytorch pytorch 25065 gh/zdevito/93/head
